custom/icds_reports/static/js/angular-services/date-helper.service.js

```diff
@@ -81,6 +104,15 @@ window.angular.module('icdsApp').factory('dateHelperService', ['$location', func
     function getReportStartDates() {
         return reportStartDates;
     }
+    function isBetweenFirstAndThirdDayOfCurrentMonth(date) {
+        var currentDate = new Date();
+        return (date < new Date(currentDate.getFullYear(), currentDate.getMonth(), 3)) &&
+            (date >= new Date(currentDate.getFullYear(), currentDate.getMonth(), 1));
+    }
+    function getPreviousMonthDate() {
+        var currentDate = new Date();
+        return new Date(currentDate.getFullYear(), (currentDate.getMonth() - 1));
```

Comment:
I don't think this works for January, does it? Is that handled elsewhere?

Reply:
If it is january, december of previous year will be returned.

![Screenshot from 2020-04-22 12-25-01](https://user-images.githubusercontent.com/56248457/79950230-60116380-8494-11ea-86d2-717e1bc223ae.png)

Reply:
wow javascript date behavior never ceases to amaze me